Work

ritasumtsova opened the pull request:
I've created 4 empty pages Login with fake auth, Departments, Departments and Error.
And added React Router 6 including lazy routes and PrivateRoute component.

ritasumtsova committed:
added React Router, PrivateRoute component and 4 main pages

ritasumtsova committed:
small PrivateRoute fixes

ritasumtsova committed:
eslint fixes

didevgen commented on package.json:
```diff
@@ -10,10 +10,12 @@
     "react-dom": "^18.1.0",
+    "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
-    "typescript": "^4.6.3",
```
please check ts present in package.json

ritasumtsova replied:
my bad

didevgen commented on src/App.tsx:
```diff
@@ -1,26 +1,43 @@
+    <Routes>
+      <Route path="/" element={<Login />} />
+      <Route element={<PrivateRoute />}>
+        <Route
+          path="/departments/:id"
+          element={
+            (
+              <Suspense fallback={<p>Loading...</p>}>
+                <Department />
+              </Suspense>
+            )
+            }
+        />
```
could you please check against that example?
https://stackoverflow.com/a/70528778

ritasumtsova replied:
sure, thank you

ritasumtsova replied:
Checked this one, works as expected. I used this syntax to fix an error

**Uncaught Error: [undefined] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>**

It happens when I wrap all my lazy routes in one Suspense component

didevgen replied:
try wrapping with a fragment as well

ritasumtsova replied:
I've tried but got the same error.

Also I've seen the same syntax on the React Router [docs](https://stackblitz.com/github/remix-run/react-router/tree/main/examples/lazy-loading?file=src/App.tsx) so I've decided I can write code like that

didevgen replied:
yeah have seen that too

ritasumtsova committed:
added typescript to package.json, auth.ts and departments.ts API files, types dir, api calls to the Login, Departments and Department components

ritasumtsova commented:
I've added:
1. TS to the package.json
2. API dir with auth.ts and departments.ts files
3. API calls to the Login, Department, Departments components
4. types dir with the auth, departments, department, employees files

ritasumtsova committed:
remove comments

didevgen commented on src/components/pages/Department/Department.tsx:
```diff
@@ -0,0 +1,42 @@
+  const { id } = useParams();
+  const [employees, setEmployees] = useState<IEmployee[]>();
+
+  const getDepartmentInfo = async () => {
+    if (id) {
+      await DepartmentsAPI.getDepartmentInfo(id)
+        .then((res) => {
+          setEmployees(res.data.employees);
+        })
+        .catch((err) => console.log(err));
+    }
+  };
```
either move outside of function either memoize

didevgen replied:
and bad practice to use `await` among with `then`

didevgen commented on src/components/pages/Departments/Departments.tsx:
```diff
@@ -0,0 +1,41 @@
+const Departments: React.FC = () => {
+  const [departments, setDepartments] = useState<IDepartments>();
+
+  const getDepartments = async () => {
```
same here

didevgen commented on src/components/pages/Departments/Departments.tsx:
```diff
@@ -0,0 +1,41 @@
+import { IDepartments } from '../../../types/departments';
+
+const Departments: React.FC = () => {
+  const [departments, setDepartments] = useState<IDepartments>();
```
set default value as empty array

didevgen commented on src/routes/PrivateRoute.tsx:
```diff
@@ -0,0 +1,10 @@
+import { Navigate, Outlet } from 'react-router-dom';
+
+const PrivateRoute: React.FC = () => {
+  return !localStorage.getItem('TOKEN')
```
TOKEN should be in shared constant

ritasumtsova committed:
fixed comments, added useMemo hook to the Departments and Department component, changed async func syntax to try catch instead of then

ritasumtsova commented:
Hopefully I followed your point. I fixed the comments, add useMemo hook to the Departments and Department component and changed async func syntax to try catch.

ritasumtsova commented:
Right, see you. Such a silly mistake.
Should I keep useMemo to better performance or this one is unnecessary here?

didevgen commented:
of course

ritasumtsova committed:
added the default [] to the useState

ritasumtsova committed:
small fix

ritasumtsova commented:
Added an empty array as a default value to the useState

didevgen commented on src/components/pages/Department/Department.tsx:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <>
+      <h1>Department page</h1>
+      {employees
+        ? employees.map((employee) => {
```
that one is always true now :)

ritasumtsova replied:
Jeez, thank you :)

ritasumtsova replied:
fixed this one

ritasumtsova committed:
fixed render condition in the Departments and Department component

ritasumtsova commented:
🥳🥳
